crates/agent/src/publications/specs.rs

```diff
@@ -300,6 +300,41 @@ async fn update_live_specs(
     Ok(lock_failures)
 }
 
+pub async fn check_source_capture_annotations(
+    draft: &tables::DraftCatalog,
+    pool: &sqlx::PgPool,
+) -> anyhow::Result<tables::Errors> {
+    let mut by_image: BTreeMap<String, bool> = BTreeMap::new();
+    let mut errors = tables::Errors::default();
+
+    for materialization in draft.materializations.iter() {
+        let Some(model) = materialization.model() else { return Ok(errors) };
+        let Some(image) = model.connector_image() else { return Ok(errors) };
+        let (image_name, image_tag) = split_image_tag(image);
+        let Some(connector_spec) = agent_sql::connector_tags::fetch_connector_spec(&image_name, &image_tag, pool).await? else { return Ok(errors) };
```

Comment:
If there's no `connector_tags` row, then we just silently skip this validation? That seems wrong, no? I could see how it would be pretty inconvenient to error in cases where we temporarily switch a task to use a tag from a PR branch or something. But IMO it seems pretty reasonable to temporarily remove the `sourceCapture` in that case. So I'm not really seeing a reason to skip validation. WDYT?

Reply:
@psFried there is already a check that checks for validity of the connector tag, and these two checks run together, so I've not included any more connector validations here, otherwise we will be validating connector tags twice (i.e. we will essentially run `check_connector_image` in this function as well to achieve validation of connector tags).

Reply:
The issue is that `check_connector_image` only checks that the _image_ is in the `connectors` table. It doesn't check `connector_tags` at all. So it's possible that `fetch_connector_spec` to return `None`, and also for `check_connector_image` to not return any errors.

Reply:
@psFried and I think that's by design: we want to allow custom image tags to be specified in publications for testing purposes (I do that quite frequently). I don't think we should fail all publications with tags that are not in the connector_tags table

Reply:
I agree that it's really useful to test with different tags. But if we skip the validation here, then it's only going to fail in the controller, which is far less obvious. The goal of this validation is to fail fast with a clear error message, and I'd think we especially want that in testing scenarios where you're already trying to debug some other issue. It also doesn't seem like too great a burden to remove the `sourceCapture` when you switch the connector image tag.

We're also able to add `:testing` tags to `connector_tags` now, since the UI will prefer a `:v1` tag.

Reply:
@psFried Resolved this 👍🏽